tcp.c

```diff
@@ -425,38 +425,48 @@ tcp_connect(char *server)
 	int i;
 
 #ifdef IPv6
-
 	int n;
-	struct addrinfo hints, *res, *ressave;
+	struct addrinfo hints, *addr;
+	static struct addrinfo *res = NULL;
```

Comment:
It's easy to miss static local variables, so perhaps it should be a static global?